scenario-runner/my_scenario.py

```diff
@@ -0,0 +1,137 @@
+    DriveDistance,
+)
+from srunner.scenarios.basic_scenario import BasicScenario
+from srunner.scenarioconfigs.scenario_configuration import ScenarioConfiguration
+
+
+class MyScenario(BasicScenario):
+    def __init__(
+        self,
+        world,
+        ego_vehicles,
+        config: ScenarioConfiguration,
+        randomize: bool = False,
+        debug_mode: bool = False,
+        criteria_enable: bool = True,
+        timeout=60,
+    ) -> None:
+        """
+        :param world: CARLA world in which the scenario is running
+        :param list[carla.Vehicle] ego_vehicles: CARLA vehicle objects created based on scenario XML configuration
+        :param ScenarioConfiguration config: Specifications from scenario XML configuration
+        :param bool randomize:
+        :param bool debug_mode:
+        :param bool criteria_enable:
+        :param float timeout: Threshold (in seconds) after which test automatically fails
+
+        :return: None
+        :rtype: None
+        """
+        # Must be defined before super() call because BasicScenario
+        # references is in its __init__() function.
+        self.timeout = timeout
+
+        super(MyScenario, self).__init__(
+            "MyScenario",
+            ego_vehicles,
+            config,
+            world,
+            debug_mode=debug_mode,
+            criteria_enable=criteria_enable,
+        )
+
+        self.world_map = CarlaDataProvider.get_map()
+
+        self.carma_vehicle = ego_vehicles[0]
+        self.other_actors_dict = {}
+
+    def _initialize_actors(self, config: ScenarioConfiguration) -> None:
+        """
+        Note: this function overrides the one in BasicScenario (parent
+        class), so this override is responsible for adding the actors
+        defined in the scenario XML configuration.
+
+        :param ScenarioConfiguration config: Specifications from
+        scenario XML configuration
+        :return: None
+        """
+        actors = CarlaDataProvider.request_new_actors(config.other_actors)
+
+        self.other_actors_dict = {
+            actor_config.rolename: actor
+            for actor_config, actor in zip(config.other_actors, actors)
+        }
+
+    def _setup_scenario_trigger(self, _: ScenarioConfiguration) -> None:
+        """
+        Set up the scenario start trigger
+
+        Note: this function overrides the abstract one in the
+        BasicScenario parent class. The base class's implementation adds
+        a trigger that prevents the scenario from starting until the
+        ego vehicle drives some distance. We don't want that trigger
+        for this scenario.
+
+        :return: None
+        """
+        pass
+
+    def _create_behavior(self):
+        """
+        Setup the behavior for NewScenario
+
+        Note: this function overrides the abstract one in the
+        BasicScenario parent class.
+
+        :return: Behavior tree root
+        """
+        start_condition = Idle(5, name="start_condition")
```

Comment:
I think option 1 would be cleaner and decouple "scenario start logic" from "actual scenario logic,"  but I haven't tried it. It might also provide a way for CDASim to prevent the scenario from starting until everything is set up.

Reply:
We should definitely look into this. Kyle mentioned to me that the main issue is with CARLA Clock which will be default spin right when the CARLA executable starts up before connecting to any client and even though you can reset the CARLA world and actors apparently Kyle said he did not find a way to reset the Time. Maybe this will enable us to do just that which would be awesome.

Reply:
But of course for a later investigation.